### DISCUSSION TITLE
compare two cool balance matrices

### DISCUSSION THREAD

**Discussion by BenxiaHu** [open2c/cooler#378] [Q&A] [answered]
Hello,
   is there any idea to calculate the difference between two cool balance matrices? I know hicCompareMatrices can (https://hicexplorer.readthedocs.io/en/latest/content/tools/hicCompareMatrices.html) can do that.

**Comment by sergpolly** [comment, marked as answer]
There is no such functionality built-in into cooler or cooltools by itself. However you can easily do this while viewing your data interactively in higlass (or using [higlass-python](https://github.com/higlass/higlass-python) locally):
<img width="1675" alt="Screenshot 2024-01-18 at 7 30 48 PM" src="https://github.com/open2c/cooler/assets/6790270/3095c34d-eca7-4c78-8ef3-b95c05af3884">
note that "divide by" button in the menu - here is the result after the division by one of the available hg19 datasets and changing a colormap to "bwr":
<img width="1645" alt="Screenshot 2024-01-18 at 7 31 44 PM" src="https://github.com/open2c/cooler/assets/6790270/f7033700-f3fc-4203-92d4-6529bdf888fc">

You can also achieve that by scripting it in python , e.g.:
```python
import cooler
clr1 = cooler.Cooler("cooler1.cool")
clr2 = cooler.Cooler("cooler2.cool")

region_of_interest = ("chr1", 10_000_000, 50_000_000)

mat1 = clr1.matrix(balance=True).fetch(region_of_interest)
mat2 = clr2.matrix(balance=True).fetch(region_of_interest)

ratio_mat = mat1 / mat2
```
now you have your ratio_mat - and can display it e.g. following this tutorial https://github.com/open2c/open2c_examples/blob/master/viz.ipynb or just plotting it using matplotlib e.g. like here: https://www.geeksforgeeks.org/imshow-with-two-colorbars-under-matplotlib/

**Reply by oganesson123** [reply]
Hello, I have two samples. I have obtained the .matrix and .bed files for these samples via hicpro, and converted these files into .cool format using hicConvertFormat.May I directly utilise the cooler balance function to normalise the .cool files for these two samples, then proceed with your comparison method? Given that these two samples possess differing sequencing depths, or rather, varying total read counts, I am uncertain whether cooler balance can adequately eliminate such disparities to permit a direct comparison of the interaction matrices between the two sample groups.

**Reply by Phlya** [reply]
After balancing the data are normally quite comparable to each other. If the sequencing depth is vastly different you can have problems (mainly coming from the different data sparsity), but in my experience the differences in library quality (cis/trans, different average contact frequency by diagonal) are typically a larger issue than somewhat unequal sequencing depth.
